operator/pkg/resources/listener_config.go

```diff
@@ -0,0 +1,192 @@
+)
+
+const (
+	// DefaultExternalKafkaListenerName is the default name of a kafka external listener
+	DefaultExternalKafkaListenerName = vectorizedv1alpha1.ExternalListenerName
+	// InternalListenerName is the name of the internal Kafka listener
+	InternalListenerName = vectorizedv1alpha1.InternalListenerName
+	// ExternalListenerName is the default name of external listener
+	ExternalListenerName = DefaultExternalKafkaListenerName
+	// DefaultExternalPandaproxyListenerName is the default name of a pandaproxy external listener
+	DefaultExternalProxyListenerName = vectorizedv1alpha1.PandaproxyExternalListenerName
+	// InternalProxyListenerName is the name of the internal proxy listener
+	InternalProxyListenerName = "proxy"
+	// DefaultExternalSchemaRegistryListenerName is the default name of a schema registry external listener
+	DefaultExternalSchemaRegistryListenerName = vectorizedv1alpha1.SchemaRegistryExternalListenerName
+
+	// KafkfaAPIConfigPath is the path to the kafka API configuration at the RP brokers.
+	KafkaAPIConfigPath = "redpanda.kafka_api"
+	// AdvertisedKafkaAPIConfigPath is the path to the advertised kafka API configuration at the RP brokers.
+	AdvertisedKafkaAPIConfigPath = "redpanda.advertised_kafka_api"
+	// PandaproxyAPIConfigPath is the path to the pandaproxy API configuration at the
+	PandaproxyAPIConfigPath = "pandaproxy.pandaproxy_api"
+	// AdvertisedPandaproxyAPIConfigPath is the path to the advertised pandaproxy API configuration at the RP brokers.
+	AdvertisedPandaproxyAPIConfigPath = "pandaproxy.advertised_pandaproxy_api"
+	// SchemaRegistryAPIConfigPath is the path to the schema registry API configuration at the RP brokers.
+	SchemaRegistryAPIConfigPath = "schema_registry.schema_registry_api"
+	// KafkaAPIConfigTLSPath is the path to the kafka API TLS configuration at the RP brokers.
+	KafkaAPIConfigTLSPath = "redpanda.kafka_api_tls"
+	// PandaproxyAPIConfigTLSPath is the path to the pandaproxy API TLS configuration at the RP brokers.
+	PandaproxyAPIConfigTLSPath = "pandaproxy.pandaproxy_api_tls"
+	// SchemaRegistryAPIConfigTLSPath is the path to the schema registry API TLS configuration at the RP brokers.
+	SchemaRegistryAPIConfigTLSPath = "schema_registry.schema_registry_api_tls"
+)
+
+// AdditionalListenerCfgNames contains the list of the listener names supported in additionalConfiguration.
+var AdditionalListenerCfgNames = []string{
+	KafkaAPIConfigPath,
+	AdvertisedKafkaAPIConfigPath,
+	PandaproxyAPIConfigPath,
+	AdvertisedPandaproxyAPIConfigPath,
+	SchemaRegistryAPIConfigPath,
+	KafkaAPIConfigTLSPath,
+	PandaproxyAPIConfigTLSPath,
+	SchemaRegistryAPIConfigTLSPath,
+}
+
+// TemplateInt defines for a template used to compute an integer value, like port {{39002 | add .Index}}
+type TemplatedInt string
+
+func (t TemplatedInt) MarshalJSON() ([]byte, error) {
+	return []byte(fmt.Sprintf(`"removequote%sremovequote"`, t)), nil
```

Comment:
Please extract the `removequote` into a `const` and explain why it's necessary in a comment.